### PR TITLE
Fixed bug in link when passing numpy int types in shape.

### PR DESCRIPTION
**Pull request by bkvogel**
This fixes a bug when supplying a shape to the `__init__()` of link that contains numpy integer types. For example a shape of the form `(numpy.int64, int)` should be considered a valid shape.

### CODE REVIEW

File: chainer/link.py [hunk 1]
```diff
@@ -13,13 +13,18 @@
 def _is_shape(value):
     if value is None:
         return True
-    elif isinstance(value, six.integer_types):
-        return True
     elif isinstance(value, collections.Sequence):
-        return all(isinstance(x, six.integer_types) for x in value)
-    else:
+        try:
+            return all(isinstance(int(x), six.integer_types) for x in value)
```
Comment (okuta) [inline chainer/link.py]: Please remove `isinstance`.
`int` function returns integer type value.

File: chainer/link.py [hunk 2]
```diff
@@ -13,13 +13,18 @@
...
+        except TypeError:
+            return False
+    try:
+        return isinstance(int(value), six.integer_types)
```
Comment (okuta) [inline chainer/link.py]: ditto

Comment (bkvogel) [inline chainer/link.py]: Thanks for the suggestion. I have made the change.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**